api/test/OpenTelemetry/Logging/CoreSpec.hs

```diff
@@ -7,24 +7,33 @@ import qualified Data.HashMap.Strict as H
 import Data.IORef
 import qualified OpenTelemetry.Attributes as A
 import OpenTelemetry.Internal.Logging.Types
+import OpenTelemetry.Internal.Logging.Types (LoggerProvider (loggerProviderProcessors))
 import qualified OpenTelemetry.LogAttributes as LA
 import OpenTelemetry.Logging.Core
 import OpenTelemetry.Resource
 import OpenTelemetry.Resource.OperatingSystem
 import Test.Hspec
 
 
+newtype TestLogProcessor body = TestLogProcessor (LogProcessor body)
+
+
+instance Show (TestLogProcessor body) where
+  show _ = "LogProcessor {..}"
```

Comment:
Added because `shouldSatisfy` has a `Show` constraint but `(Show LogProcessor)` does not make sense because of the functions that are properties. `"LogProcessor {..}"` is used to show that there are internal properties that exist but are not shown.